### PR TITLE
test memory leakage

### PR DESCRIPTION
**Pull request by sasurobert**
resolved memory leakage when using Cache from a wasmer instance

### CODE REVIEW

**Review by andrei-marinica — APPROVED** [review]

File: wasmer/instance.go [hunk 1]
```diff
@@ -220,6 +222,13 @@ func NewInstanceFromCompiledCodeWithOptions(
 		c_instance_context := cWasmerInstanceContextGet(c_instance)
 		instance.InstanceCtx = IntoInstanceContextDirect(c_instance_context)
 	}
+
+	hackMap := make(map[unsafe.Pointer]interface{})
```
Comment (AdoAdoAdo) [inline wasmer/instance.go:226]: why is the hackMap needed?
aren't just lines 206 and 230 relevant?

File: wasmer/instance.go [hunk 2]
```diff
@@ -280,8 +289,12 @@ func (instance *Instance) Cache() ([]byte, error) {
 	}
 
 	goBytes := C.GoBytes(unsafe.Pointer(cacheBytes), C.int(cacheLen))
-	cacheBytes = nil
 
+	hackMap := make(map[unsafe.Pointer]interface{})
```
Comment (iulianpascalau) [inline wasmer/instance.go:293]: here too? Why do we need the hack map?

Comment (sasurobert) [reply wasmer/instance.go:293]: testing now - as without it is fails in some scenarios.

File: wasmer/instance.go (hunk 1 above)
Comment (sasurobert) [reply wasmer/instance.go:226]: testing now - as without it is fails in some scenarios.

**Review by dorin-iancu — DISMISSED** [review]

File: wasmer/instance.go [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package wasmer
 
+// #include <stdlib.h>
```
Comment (iulianpascalau) [inline wasmer/instance.go:3]: please remove comment

**Review by iulianpascalau — COMMENTED** [review]
Please add description

File: wasmer/instance.go (hunk 3 above)
Comment (sasurobert) [reply wasmer/instance.go:3]: this is not a comment. this is a c include - it is a must for C.free

Comment (iulianpascalau) [reply wasmer/instance.go:3]: Ok

**Review by AdoAdoAdo — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**